rts/Sim/Units/Unit.cpp

```diff
@@ -662,6 +662,22 @@ void CUnit::Update()
 	if (isDead)
 		return;
 
+	 // --- NEW Self-Destruct Check ---
+	if (selfDTargetFrame > 0) {
+		if (gs->frameNum >= selfDTargetFrame) {
+			// avoid unfinished buildings making an explosion
+			KillUnit(nullptr, !beingBuilt, beingBuilt, -CSolidObject::DAMAGE_SELFD_EXPIRED);
+		return; // Skip rest of update if killed
+		} else {
+			const float remainingSeconds = std::max(0.0f, (static_cast<float>(selfDTargetFrame) - gs->frameNum) / GAME_SPEED);
```

Comment:
Why the std::max, can targetFrame ever exceed gs->FrameNum if this is done in Update and not SlowUpdate?

The casts (assuming std::max is kept) can also be simplified via
```suggestion
			const float remainingSeconds = std::max(0.0f, (selfDTargetFrame - gs->frameNum) * INV_GAME_SPEED);
```

Reply:
No I suppose it can't, I'll update it